Drop macOS x86_64 to tier 2

mtreinish opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
-->

### Summary

With Apple deprecating the platform support and removing it in future OS versions, github is similarly marking the end of support for x86_64 macOS suppport in github actions:

https://github.blog/changelog/2025-09-19-github-actions-macos-13-runner-image-is-closing-down/

With this our hands our tied as we rely on github actions to test and build on the platform in CI. As the platform is still supported for now this commit starts the process of dropping it as a supported platform by downgrading it to tier 2. This means that we are no longer running tests on x86_64 macOS in PR CI jobs, it is relegated to only run during the wheel builds. This commit makes the necessary CI configuration changes to do this and adds a release note to document the downgrade of support. We should update the platform support docs page at:

https://quantum.cloud.ibm.com/docs/en/guides/install-qiskit#operating-system-support

when we are preparing the 2.3.0 release to indicate this change there too.

### Details and comments

mtreinish committed:
Drop macOS x86_64 to tier 2

With Apple deprecating the platform support and removing it in future OS
versions, github is similarly marking the end of support for x86_64
macOS suppport in github actions:

https://github.blog/changelog/2025-09-19-github-actions-macos-13-runner-image-is-closing-down/

With this our hands our tied as we rely on github actions to test and
build on the platform in CI. As the platform is still supported for now
this commit starts the process of dropping it as a supported platform by
downgrading it to tier 2. This means that we are no longer running tests
on x86_64 macOS in PR CI jobs, it is relegated to only run during the
wheel builds. This commit makes the necessary CI configuration changes
to do this and adds a release note to document the downgrade of
support. We should update the platform support docs page at:

https://quantum.cloud.ibm.com/docs/en/guides/install-qiskit#operating-system-support

when we are preparing the 2.3.0 release to indicate this change there
too.

jakelishman commented on .github/workflows/wheels-build.yml:
```diff
@@ -73,7 +73,7 @@ jobs:
           - ubuntu-latest
           # Used for the x86_64 builds.
-          - macos-13
+          - macos-15-intel
```
Does this runner exist? I couldn't see its docs at https://github.com/actions/runner-images/

mtreinish replied:
It does, see the post I linked in the commit message, github refers to it as the path to keep testing with macos x86_64.

mtreinish committed:
Move macos-intel job to second wheel stage

The previous commit left the macos x86_64 build in the first wheel build
stage. This however meant we were treating the job as a tier 1 platform
build still and also running PGO on the build. Since we reserve those
for tier 1 platforms it wasn't a good fit. Also, from the single test run
the new macOS 15 intel runner is signficantly slower (which is not
definitive evidence due to the variability of cloud hosting performance)
so blocking the tier 1 platforms on the macOS intel build isn't
desireable. This commit fixes this by moving it to the second stage of
the build and disabling PGO for that build.

mtreinish commented:
I've pushed up a docs PR: https://github.com/Qiskit/documentation/pull/3951 to update the platform support docs to reflect this change. We won't be able to merge that until we update the docs for 2.3.0 though.

jakelishman commented on .github/workflows/wheels-build.yml:
```diff
@@ -73,7 +81,6 @@ jobs:
         os:
           - ubuntu-latest
           # Used for the x86_64 builds.
```
```suggestion
```

jakelishman commented on releasenotes/notes/macos-x86_64-tier-2-023a00d34984665f.yaml:
```diff
@@ -0,0 +1,16 @@
+
+    While platform support hasn't been removed yet it is worth pointing out that this
+    will likely happen in the lead up to Fall 2027 when Github has documented that they
+    will retire the only images with support for running on x86_64 macOS. [#]_
+
+    .. [#] https://github.blog/changelog/2025-09-19-github-actions-macos-13-runner-image-is-closing-down/
```
hyperlink

mtreinish replied:
Did you want me to replace the footnote with an inline hyperlink? The footnote here should automatically be a link since sphinx should detect the url and automatically make it a link.

jakelishman replied:
I don't remember what we were talking about with respect to this.

mtreinish committed:
Update .github/workflows/wheels-build.yml

Co-authored-by: Jake Lishman <jake@binhbar.com>
